test/cases/uri_gid_test.rb

```diff
@@ -45,6 +45,28 @@ class URI::GIDTest <  ActiveSupport::TestCase
   end
 end
 
+class URI::GIDIDEncodingTest < ActiveSupport::TestCase
+  test 'alphanumeric' do
+    model = Person.new('John123')
+    assert_equal 'gid://app/Person/John123', URI::GID.create('app', model).to_s
+  end
+
+  test 'non-alphanumeric' do
+    model = Person.new('John Doe-Smith/Jones')
```

Comment:
See this is what worried me about the original issue. This doesn't exactly feel like a proper `model_id` to me.